pkg/providers/aws/cluster_network_provider.go

```diff
@@ -685,6 +686,63 @@ func (n *NetworkProvider) IsEnabled(ctx context.Context) (bool, error) {
 	return len(validBundledVPCSubnets) == 0, nil
 }
 
+// DeleteBundledCloudResources returns an error on any error deleting of the following resources
+// * elasticache subnet group
+// * rds subnet group
+// * ec2 security group
+//
+// it has been located under the cluster network provider as it requires 3 different aws sessions
+// (elasticache, rds and ec2) to delete the required resources even though it deals with bundled
+// resources. The majority of the functionality in this file relates to standalone aws vpc and it's
+// resources.
+func (n *NetworkProvider) DeleteBundledCloudResources(ctx context.Context) error {
+	logger := n.Logger.WithField("action", "deleteBundledCloudResources")
+
+	subnetGroupName, err := BuildInfraName(ctx, n.Client, "subnetgroup", DefaultAwsIdentifierLength)
+	if err != nil {
+		return errorUtil.Wrap(err, "error building bundle subnet group resource name on deletion")
+	}
+	logger.Info("deleting bundled elasticcache subnet group %s , if it's not found it's already deleted and will continue", subnetGroupName)
+	_, err = n.ElasticacheApi.DeleteCacheSubnetGroup(&elasticache.DeleteCacheSubnetGroupInput{
+		CacheSubnetGroupName: aws.String(subnetGroupName),
+	})
+	if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() != "CacheSubnetGroupNotFoundFault" {
+		return errorUtil.Wrap(err, "error deleting elasticache subnet group")
+	}
+	logger.Info("deleting bundled rds subnet group %s , if it's not found it's already deleted and will continue", subnetGroupName)
+	_, err = n.RdsApi.DeleteDBSubnetGroup(&rds.DeleteDBSubnetGroupInput{
+		DBSubnetGroupName: aws.String(subnetGroupName),
+	})
+	if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() != "DBSubnetGroupNotFoundFault" {
+		return errorUtil.Wrap(err, "error deleting rds subnet group")
+	}
+
+	securityGroupName, err := BuildInfraName(ctx, n.Client, "securitygroup", DefaultAwsIdentifierLength)
+	if err != nil {
+		return errorUtil.Wrap(err, "error building bundle security group resource name on deletion")
+	}
+	logger.Info("Deleting bundled ec2 security group %s , if it's not found it's already deleted and will continue", securityGroupName)
+	// in the case of the security group the Group Id is required in order to delete security groups
+	// not connected with the default vpc. In order to delete it it is required to describe them
+	// all in the account and then find the one with the correct group name and then request deletion
+	// using the group id of the matched security group
+	securityGroupOutput, err := n.Ec2Api.DescribeSecurityGroups(&ec2.DescribeSecurityGroupsInput{})
+	if err != nil {
+		return errorUtil.Wrap(err, "error getting ec2 security group")
+	}
+	for _, securityGroup := range securityGroupOutput.SecurityGroups {
+		if *securityGroup.GroupName == securityGroupName {
```

Comment:
Should we be checking that this security group is targeting the bundled vpc? Just to ensure we are deleting the correct security group and potentially avoid a similar issue we seen last Friday

Reply:
Realistically that should never happen. But checking it targets the bundled VPC makes sense for dev environments